packages/strapi-plugin-users-permissions/admin/src/containers/AuthPage/index.js

```diff
@@ -40,73 +41,114 @@ import styles from './styles.scss';
 
 export class AuthPage extends React.Component { // eslint-disable-line react/prefer-stateless-function
   componentDidMount() {
-    const params = this.props.location.search ? replace(this.props.location.search, '?code=', '') : this.props.match.params.id;
-    this.props.setForm(this.props.match.params.authType, params);
+    auth.clearAppStorage();
+    this.setForm();
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.match.params.authType !== nextProps.match.params.authType) {
-      const params = nextProps.location.search ? replace(nextProps.location.search, '?code=', '') : nextProps.match.params.id;
-      this.props.setForm(nextProps.match.params.authType, params);
-      this.props.hideLoginErrorsInput(false);
+  componentDidUpdate(prevProps) {
+    const { 
+      hideLoginErrorsInput,
+      match: { 
+        params : {
+          authType,
+        },
+      }, 
+      submitSuccess,
+    } = this.props;
+
+    if (authType !== prevProps.match.params.authType) {
+      this.setForm();
+      hideLoginErrorsInput(false);
     }
-
-    if (nextProps.submitSuccess) {
-      switch (this.props.match.params.authType) {
+    if (submitSuccess && submitSuccess !== prevProps.submitSuccess) {
```

Comment:
`if (submitSuccess) { //... }` is enough as a check here.

(Plus there's an extra line missing before the condition 😉)